fiat-github/src/main/java/com/netflix/spinnaker/fiat/roles/github/GithubTeamsUserRolesProvider.java

```diff
@@ -115,52 +116,24 @@ private boolean isMemberOfOrg(String username) {
     return isMemberOfOrg;
   }
 
-  private List<Team> getTeams() {
-    List<Team> teams = new ArrayList<>();
-    int page = 1;
-    boolean hasMorePages = true;
-
-    do {
-      List<Team> teamsPage = getTeamsInOrgPaginated(page++);
-      teams.addAll(teamsPage);
-      if (teamsPage.size() != gitHubProperties.paginationValue) {
-        hasMorePages = false;
-      }
-      log.debug("Got " + teamsPage.size() + " teams back. hasMorePages: " + hasMorePages);
-    } while (hasMorePages);
-
-    return teams;
-  }
+  private List<Team> getTeamsGraphql(String username) {
+    log.debug("Fetching " + username + " teams.");
+    List<TeamEdge> edges = new ArrayList<>();
 
-  private List<Team> getTeamsInOrgPaginated(int page) {
-    List<Team> teams = new ArrayList<>();
     try {
-      log.debug("Requesting page " + page + " of teams.");
-      teams = gitHubClient.getOrgTeams(gitHubProperties.getOrganization(),
-                                       page,
-                                       gitHubProperties.paginationValue);
+      GraphqlTeamRequest request = new GraphqlTeamRequest();
+      request.setQuery(buildGraphqlQuery(gitHubProperties.getOrganization(), username));
+      edges = gitHubClient.getUserTeams(request).getData().getOrganizationResult().getTeams().getEdges();
```

Comment:
To prevent any of these accessors being `null` and causing an NPE, I suggest making each one of your data classes below have default, non-null values.